### PR TITLE
New Mining Loot

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
All of the below are found in necropolis chests.

**Wisp Lantern:**

sprites

Bit blurry since it's orbiting, but you get the idea. It pulses as well. 

You open the lantern and the wisp orbits your head, providing light and letting you see mobs through walls. 

Sprites are from my girlfriend.

**Red/Blue Cubes**
Using one warps you to the other.

**Meat Hook**

Fire it at people, pull them to your location.

Sprites are from my girlfriend.

**Immortality Talisman**
Using it renders you invulnerable and immobile for 10 seconds. 60 second cooldown.

Sprites by @WJohn

**Paradox Bag**

A bag that exists in two places at once. It comes with a copy of itself, and two people can access/share the same bagspace by wearing a copy each.

:cl: Kor
rscadd: Added wisp lanterns to necropolis chests.
rscadd: Added red/blue cube pairs to necropolis chests.
rscadd: Added meat hooks to necropolis chests.
rscadd: Added immortality talismans to necropolis chests.
rscadd: Added paradox bags to necropolis chests.
/:cl:

### CODE REVIEW

**Comment by Armhulen** [comment]
Cool!

**Comment by RemieRichards** [comment]
@KorPhaeron where are these supposed comments? there literally aren't any.

>I've left comments on the offending lines.

**Comment by KorPhaeron** [comment]
@RemieRichards 

Clever as I am, I wrote the comments but did not save/commit them

File: code/modules/mining/abandoned_crates.dm [hunk 1]
```diff
@@ -287,3 +287,48 @@
...
+		user << "You release the wisp. It begins to bob around your head."
+		wisp.orbit(user, 20)
+		user.sight |= SEE_MOBS
+		icon_state = "lantern"
+	else
+		if(wisp.orbiting)
+			var/atom/A = wisp.orbiting
+			if(istype(A, /mob/living))
+				var/mob/living/M = A
+				M.sight -= SEE_MOBS // This doesn't work
```
Comment (RemieRichards) [inline code/modules/mining/abandoned_crates.dm]: sight is a bitflag. so this needs to be ```&= ~SEE_MOBS```

File: code/modules/mining/abandoned_crates.dm [hunk 2]
```diff
@@ -287,3 +287,48 @@
...
+				M << "The wisp has returned to it's latern. Your vision returns to normal." //This works
+			wisp.stop_orbit() //This works
+		wisp.loc = src //This works
+		user << "You return the wisp to the latern."
+		icon_state = "lantern-blue"//This doesn't work
```
Comment (RemieRichards) [inline code/modules/mining/abandoned_crates.dm]: define "doesn't work", does the icon turn blank? if so you typo'd the state.

Comment (KorPhaeron) [inline code/modules/mining/abandoned_crates.dm]: Icon_state stays as "lantern"

Comment (RemieRichards) [inline code/modules/mining/abandoned_crates.dm]: that's illogical, this line CLEARLY sets the ```icon_state``` to ```"lantern-blue"```, and if the lines above it work, this line has to work, ergo you've probably typo'd and I'd urge you to check.

File: code/modules/mining/abandoned_crates.dm [hunk 3]
```diff
@@ -287,3 +287,48 @@
...
+	W.home = src
+
+/obj/effect/wisp
+	name = "friendly wisp"
+	desc = "Happy to light your way."
+	icon = 'icons/obj/lighting.dmi'
+	icon_state = "orb"
+	var/obj/item/device/wisp_lantern/home
+	luminosity = 7
+	layer = 5
```
Comment (ChangelingRain) [inline code/modules/mining/abandoned_crates.dm]: MOB_LAYER + 0.7 or FLY_LAYER - 0.3
above mine turfs and most mobs, below space vines and ghosts.

File: code/modules/mining/abandoned_crates.dm [hunk 4]
```diff
@@ -287,3 +287,48 @@
...
+			if(istype(A, /mob/living))
+				var/mob/living/M = A
+				M.sight -= SEE_MOBS // This doesn't work
+				M << "The wisp has returned to it's latern. Your vision returns to normal." //This works
+			wisp.stop_orbit() //This works
+		wisp.loc = src //This works
+		user << "You return the wisp to the latern."
+		icon_state = "lantern-blue"//This doesn't work
+
+/obj/item/device/wisp_lantern/New()
```
Comment (RemieRichards) [inline code/modules/mining/abandoned_crates.dm]: as I said before these need a ```Destroy()``` too

File: code/modules/mining/abandoned_crates.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/mining/abandoned_crates.dm]: You'd think it'd turn invisible then, right? If it was a typo. That's why I'm confused.

Comment (RemieRichards) [inline code/modules/mining/abandoned_crates.dm]: Yes as an invalid ```icon_state``` is usually rendered blank (as it doesn't exist)

File: code/modules/mining/abandoned_crates.dm [hunk 5]
```diff
@@ -287,3 +287,48 @@
...
 			new /obj/item/weapon/spellbook/oneuse/smoke(src)
 
+
+//Spooky special loot
+
+/obj/item/device/wisp_lantern
+	name = "spooky lantern"
+	desc = "This lantern gives off no light, but is home to a friendly wisp."
+	icon = 'icons/obj/lighting.dmi'
+	icon_state = "lantern-blue"
```
Comment (ChangelingRain) [inline code/modules/mining/abandoned_crates.dm]: If it was a typo, though, this wouldn't work(also I checked the file, not a typo?????)

File: code/modules/mining/abandoned_crates.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/mining/abandoned_crates.dm]: Okay so it

1. Starts blue

2. Click it inhand, release wisp. It stays blue.

3. Click inhand, recall wisp. It turns off.

4. Click inhand, release wisp. It stays off.

File: code/modules/mining/abandoned_crates.dm [hunk 6]
```diff
@@ -287,3 +287,48 @@
...
+
+/obj/item/device/wisp_lantern/attack_self(mob/user)
+	if(!wisp)
+		user << "The wisp has gone missing!"
+		return
+	if(wisp.loc == src)
+		user << "You release the wisp. It begins to bob around your head."
+		wisp.orbit(user, 20)
+		user.sight |= SEE_MOBS
+		icon_state = "lantern"
```
Comment (RemieRichards) [inline code/modules/mining/abandoned_crates.dm]: >Click it inhand, release wisp. It stays blue.

so this line is broken also?

Comment (KorPhaeron) [inline code/modules/mining/abandoned_crates.dm]: Correct

File: code/modules/mining/abandoned_crates.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/mining/abandoned_crates.dm]: Still doesn't work. This is very strange.

**Comment by KorPhaeron** [comment]
It needed a spawn() before the orbit started

**Comment by RemieRichards** [comment]
course it did, orbit never ends dummy.

**Comment by KorPhaeron** [comment]
See, it was both orbit related and not your fault

File: code/modules/mining/abandoned_crates.dm [hunk 7]
```diff
@@ -287,3 +287,59 @@
...
+	icon_state = "lantern-blue"
+	var/obj/effect/wisp/wisp
+
+/obj/item/device/wisp_lantern/attack_self(mob/user)
+	if(!wisp)
+		user << "The wisp has gone missing!"
+		return
+	if(wisp.loc == src)
+		user << "You release the wisp. It begins to bob around your head."
+		spawn()
```
Comment (ChangelingRain) [inline code/modules/mining/abandoned_crates.dm]: why don't you just,,, move the sight/icon changes above this?

Comment (KorPhaeron) [inline code/modules/mining/abandoned_crates.dm]: Hmmmmmmmmmmmmmmmmmmmmmmmmmmm

**Comment by Fox-McCloud** [comment]
This is pretty cute.

:+1:  I like it.

**Comment by Lumipharon** [comment]
Neat.

**Comment by coiax** [comment]
Ten seconds of invulnerable stasis every sixty seconds? Neat. I wonder what the powergamers would use it for.

**Comment by Jazaen** [comment]
Does it give you singularity protection as well? And gibbing protection? Because you could get hardsuit, get a bomb (preferably 2-bluespace beaker grenade), prime it in one hand and enable invu in other.

**Comment by KorPhaeron** [comment]
@Jazaen 

That would be possible, yeah. But that requires you to get large explosives as a miner or somehow pry this away from a miner as a scientist, and even then it wouldn't be that much better than just dropping a bag, walking away, and signalling it.

**Comment by coiax** [comment]
It puts the user in a special non-deletable godmode effect, so I'm pretty sure you could prime a bomb and then activate your talisman. Should point out that this is one of twenty five different loots you can get, so it's not gonna be even a reliable drop.

Tbh, if you've got one, you've probably earned it.

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+	..()
+	if(!linked)
+		var/obj/item/device/warp_cube/blue = new(src.loc)
+		linked = blue
+		blue.linked = src
+
+//Meat Hook
+/obj/item/weapon/gun/magic/hook
+	name = "meat hook"
+	desc = "Mid or feed."
```
Comment (Xhuis) [inline code/modules/mining/lavaland/necropolis_chests.dm]: nice

**Comment by KorPhaeron** [comment]
@ChangelingRain @Cheridan @duncathan @RemieRichards 

I think it's ready probably

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 9]
```diff
@@ -0,0 +1,336 @@
...
+	fire_sound = "sound/magic/Staff_Change.ogg"
```
Comment (ChangelingRain) [inline code/modules/mining/lavaland/necropolis_chests.dm]: duplicated; see like 170

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+	..()
+	var/obj/item/projectile/hook/P = BB
+	spawn(1)
+		P.chain = P.Beam(user,icon_state="chain",icon = 'icons/obj/lavaland/artefacts.dmi',time=1000, maxdistance = 30)
+
+/obj/item/projectile/hook/on_hit(atom/target)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		L.forceMove(get_turf(firer))
```
Comment (ChangelingRain) [inline code/modules/mining/lavaland/necropolis_chests.dm]: this being instant seems silly

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 11]
```diff
@@ -0,0 +1,336 @@
...
+
+//Meat Hook
+/obj/item/weapon/gun/magic/hook
+	name = "meat hook"
+	desc = "Mid or feed."
+	fire_sound = "sound/magic/Staff_Change.ogg"
+	ammo_type = /obj/item/ammo_casing/magic/hook
+	icon_state = "hook"
+	item_state = "chain"
+	fire_sound = 'sound/weapons/batonextend.ogg'
```
Comment (RemieRichards) [inline code/modules/mining/lavaland/necropolis_chests.dm]: that's this one!

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 10 above)
Comment (RemieRichards) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Would be much cooler to ```throw_at_fast()``` them towards the firer instead!

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 12]
```diff
@@ -0,0 +1,336 @@
...
+
+/obj/item/device/immortality_talisman/attack_self(mob/user)
+	if(cooldown < world.time)
+		cooldown = world.time + 600
+		user.visible_message("<span class='danger'>[user] vanishes from reality, leaving a a hole in their place!</span>")
+		var/obj/effect/immortality_talisman/Z = new(get_turf(src.loc))
+		Z.name = "hole in reality"
+		Z.desc = "It's shaped an awful lot like [user.name]."
+		Z.dir = user.dir
+		user.loc = Z
```
Comment (RemieRichards) [inline code/modules/mining/lavaland/necropolis_chests.dm]: ```user.forceMove(Z)``` please.

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 13]
```diff
@@ -0,0 +1,336 @@
...
+		user.notransform = 1
+		user.status_flags |= GODMODE
+		spawn(100)
+			user.status_flags &= ~GODMODE
+			user.notransform = 0
+			user.loc = get_turf(Z)
```
Comment (RemieRichards) [inline code/modules/mining/lavaland/necropolis_chests.dm]: ```user.forceMove(get_turf(Z))``` please.

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 10 above)
Comment (KorPhaeron) [inline code/modules/mining/lavaland/necropolis_chests.dm]: What if I want to do a sick fountain hook (fire the chain then hand tele out before it hits)

Comment (RemieRichards) [inline code/modules/mining/lavaland/necropolis_chests.dm]: you still can tho?

Comment (KorPhaeron) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Throwing them towards me wouldn't work if I'd teleported several rooms away though

Comment (RemieRichards) [inline code/modules/mining/lavaland/necropolis_chests.dm]: again, towards the turf the chain started at.

Comment (ChangelingRain) [inline code/modules/mining/lavaland/necropolis_chests.dm]: yeah okay throwing it then teleporting to the X is okay I guess
you can call it a reference

Comment (KorPhaeron) [inline code/modules/mining/lavaland/necropolis_chests.dm]: @ChangelingRain 

Both a throw and a teleport?

Comment (KorPhaeron) [inline code/modules/mining/lavaland/necropolis_chests.dm]: @RemieRichards 

I mean I want to fire the hook, warp, then have them pulled to where I am warped, the classic dota trick

Comment (RemieRichards) [inline code/modules/mining/lavaland/necropolis_chests.dm]: @KorPhaeron then do what I said originally then! use the firer! jeez.

Comment (KorPhaeron) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Alright it uses throw

**Comment by KorPhaeron** [comment]
@RemieRichards @ChangelingRain 

It throws now

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 14]
```diff
@@ -0,0 +1,336 @@
...
+	..()
+	var/obj/item/projectile/hook/P = BB
+	spawn(1)
+		P.chain = P.Beam(user,icon_state="chain",icon = 'icons/obj/lavaland/artefacts.dmi',time=1000, maxdistance = 30)
+
+/obj/item/projectile/hook/on_hit(atom/target)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		var/turf/T = get_turf(firer)
```
Comment (RemieRichards) [inline code/modules/mining/lavaland/necropolis_chests.dm]: why even bother defining ```T```?

Comment (PKPenguin321) [inline code/modules/mining/lavaland/necropolis_chests.dm]: i smell copypaste
i would know because i copypaste all the time

**Comment by KorPhaeron** [comment]
@RemieRichards 

aaaaa

**Comment by Chocobro** [comment]
wisp lantern is fucking adorbs

**Pull request merged**